src/main/java/net/dv8tion/jda/api/entities/MessageSticker.java

```diff
@@ -0,0 +1,146 @@
+package net.dv8tion.jda.api.entities;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class MessageSticker implements ISnowflake
+{
+    private final long id;
+    private final long packId;
+    private final String name;
+    private final String description;
+    private final String asset;
+    private final String previewAsset;
+    private final StickerFormat formatType;
+
+    public MessageSticker(final long id, final long packId, final String name, final String description, final String asset, final String previewAsset, final StickerFormat formatType)
+    {
+        this.id = id;
+        this.packId = packId;
+        this.name = name;
+        this.description = description;
+        this.asset = asset;
+        this.previewAsset = previewAsset;
+        this.formatType = formatType;
+    }
+
+    @Override
+    public long getIdLong()
+    {
+        return id;
+    }
+
+    /**
+     * The ID of the pack the sticker is from.
+     *
+     * @return the ID of the pack the sticker is from
+     */
+    public long getPackId()
+    {
+        return packId;
+    }
+
+    /**
+     * The name of the sticker.
+     *
+     * @return the name of the sticker
+     */
+    @Nonnull
+    public String getName()
+    {
+        return name;
+    }
+
+    /**
+     * The description of the sticker.
+     *
+     * @return the description of the sticker
+     */
+    @Nonnull
+    public String getDescription()
+    {
+        return description;
+    }
+
+    /**
+     * The asset hash of the sticker.
+     *
+     * @return the asset hash of the sticker
+     */
+    @Nonnull
+    public String getAsset()
```

Comment:
changed, if you have a better explanation of what the asset hash is, let me know, because i just copied this from other docs as i can't explain it myself

Reply:
Discord Docs:
> asset* | string | sticker asset hash
* The URL for fetching sticker assets is currently private.

So I would assume it's most likely the URL of the sticker, but hashed?

Reply:
The asset is the image file id(name).
Like how for emoji it uses `https://cdn.discordapp.com/emojis/$id` to get the image.
For stickers its something like `https://the private url/$asset

Reply:
At a minimum this should probably include an example of what it will look like and also include the fact that `"The URL for fetching sticker assets is currently private."`

It is possible that we might not need to surface this in JDA if it is not useable in any way due to internal discord restrictions making actionable endpoints private.